osbenchmark/results_publisher.py

```diff
@@ -145,16 +162,33 @@ def publish(self):
 
         metrics_table.extend(self._publish_transform_stats(stats))
 
+        max_throughput = -1
+        record_with_best_throughput = None
+
+        throughput_pattern = r"_(\d+)_clients$"
+
+
         for record in stats.op_metrics:
             task = record["task"]
-            metrics_table.extend(self._publish_throughput(record, task))
-            metrics_table.extend(self._publish_latency(record, task))
-            metrics_table.extend(self._publish_service_time(record, task))
-            # this is mostly needed for debugging purposes but not so relevant to end users
-            if self.show_processing_time:
-                metrics_table.extend(self._publish_processing_time(record, task))
-            metrics_table.extend(self._publish_error_rate(record, task))
-            self.add_warnings(warnings, record, task)
+            maybe_match_task_is_part_of_throughput_testing = re.search(throughput_pattern, task)
+            if maybe_match_task_is_part_of_throughput_testing:
+
+                # assumption: all units are the same and only maximizing throughput over one operation (i.e. not both ingest and search).
+                # To maximize throughput over multiple operations, would need a list/dictionary of maximum throughputs.
+                task_throughput = record["throughput"][Throughput.MEAN.value]
+                logger = logging.getLogger(__name__)
```

Comment:
Let's move this to be under the `__init__` method of SummaryResultsPublisher for when we would like to add logging elsewhere in the class.